File: app/src/main/java/anartzmuxika/connectionchangereceiver/utils/DateTime.java
package anartzmuxika.connectionchangereceiver.utils;

import java.text.DateFormat;
import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.Date;
import java.util.Locale;

/********************************************************************************
 * Created by Anartz Mugika (mugan86@gmail.com) on 20/1/18.
 ********************************************************************************/

public class DateTime {


    public static String getCurrentData ()
    {
        DateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd", Locale.ENGLISH);
        Calendar cal = Calendar.getInstance();
        return dateFormat.format(cal.getTime());

    }

    public static Calendar getCurrentDataCalendar()
    {
        return  Calendar.getInstance();
    }


    public static String getCurrentDataTime ()
    {
        DateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
        return dateFormat.format(new Date());
    }

    public static String getYesterdayData ()
    {
        DateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd", Locale.ENGLISH);
        Calendar cal = Calendar.getInstance();
        cal.add(Calendar.DAY_OF_YEAR, -1);
        return dateFormat.format(cal.getTime());
    }

    public static String [] getFirstDayNextYearAndLastDayBeforeYear ()
    {
        String [] datas = new String [2];
        DateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd", Locale.ENGLISH);
        Calendar cal = Calendar.getInstance();
        cal.add(Calendar.YEAR, 1);
        cal.set(Calendar.DAY_OF_MONTH, 1);
        cal.set(Calendar.MONTH, 0);
        datas [0] = dateFormat.format(cal.getTime());
        cal = Calendar.getInstance();
        cal.add(Calendar.YEAR, -1);
        cal.set(Calendar.DAY_OF_MONTH, 31);
        cal.set(Calendar.MONTH, 11);
        datas [1] = dateFormat.format(cal.getTime());
        return datas;
    }

    public static String getCurrentDataWithAddSetMonthValue (String first_data, int add_month)
    {
        DateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd", Locale.ENGLISH);
        Calendar cal = Calendar.getInstance();
        if (!first_data.equals("")) //add first_data in calendar object to correct asign in second data
        {
            int urtea = Integer.parseInt(first_data.substring(0, 4));
            int hilabetea = Integer.parseInt(first_data.substring(5,7))-1;
            int eguna = Integer.parseInt(first_data.substring(8,10));

            cal.set(urtea, hilabetea, eguna);
        }

        cal.add(Calendar.MONTH, add_month);  //Add two months to first data
        return dateFormat.format(cal.getTime());

    }

    public static String getCurrentDataWithAddSetBeforeDays (String first_data, int add_month)
    {
        DateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd", Locale.ENGLISH);
        Calendar cal = Calendar.getInstance();
        if (!first_data.equals("")) //add first_data in calendar object to correct asign in second data
        {
            int urtea = Integer.parseInt(first_data.substring(0, 4));
            int hilabetea = Integer.parseInt(first_data.substring(5,7))-1;
            int eguna = Integer.parseInt(first_data.substring(8,10));

            cal.set(urtea, hilabetea, eguna);
        }

        cal.add(Calendar.DAY_OF_WEEK, -(add_month));  //DownDays value to first data
        return dateFormat.format(cal.getTime());
    }

    public static Calendar getAllDataTimeCalendarObject (String date, String hour_string)
    {
        //Extract details data
        int year = Integer.parseInt(date.substring(0, 4));
        int month = Integer.parseInt(date.substring(5,7))-1;
        int day = Integer.parseInt(date.substring(8,10));

        int [] hour_int = getHourMinutesSecondsInfo(hour_string);

        System.out.println(year+"/"+month+"/"+day+" "+hour_int[0]+":"+hour_int[1]+":"+hour_int[2]);
        Calendar calendar = Calendar.getInstance();
        calendar.set(year, month, day, hour_int[0], hour_int[1], hour_int[2]);
        return calendar;
    }

    public static int [] getHourMinutesSecondsInfo (String hour_string)
    {
        String hour = hour_string.substring(0, 2);
        int hour_int;
        if (hour.charAt(0) == '0')
        {
            hour_int = Integer.parseInt(hour.substring(1));
        }
        else
        {
            hour_int = Integer.parseInt(hour);
        }

        int minute_int = Integer.parseInt(hour_string.substring(3, 5));
        return new int [] {hour_int,minute_int,0};
    }

    public static String [] setFormatData (int year, int monthOfYear, int dayOfMonth)
    {
        //year/month/day
        String [] format_data = new String [3];

        format_data[0] = String.valueOf(year);
        if (monthOfYear+1 < 10)
        {
            format_data [1] = "0"+String.valueOf(monthOfYear+1);
        }
        else
        {
            format_data [1] = String.valueOf(monthOfYear+1);
        }

        if (dayOfMonth<10)
        {
            format_data [2] = "0"+String.valueOf(dayOfMonth);
        }
        else
        {
            format_data [2] = String.valueOf(dayOfMonth);
        }
        return format_data;
    }
}

File: app/src/main/java/anartzmuxika/connectionchangereceiver/utils/DateTime.kt
package anartzmuxika.connectionchangereceiver.utils

import android.annotation.SuppressLint
import java.text.SimpleDateFormat
import java.util.Calendar
import java.util.Date
import java.util.Locale

/***********************************************
 * Created by Anartz Mugika on 22/3/14.
 * Updated in Kotlin: 20/01/2018
 ***********************************************/

object DateTime {


    val currentData: String
        get() {
            val dateFormat = SimpleDateFormat("yyyy-MM-dd", Locale.ENGLISH)
            val cal = Calendar.getInstance()
            return dateFormat.format(cal.time)

        }

    val currentDataCalendar: Calendar
        get() = Calendar.getInstance()


    val currentDataTime: String
        @SuppressLint("SimpleDateFormat")
        get() {
            val dateFormat = SimpleDateFormat("yyyy-MM-dd HH:mm:ss")
            return dateFormat.format(Date())
        }

    val yesterdayData: String
        get() {
            val dateFormat = SimpleDateFormat("yyyy-MM-dd", Locale.ENGLISH)
            val cal = Calendar.getInstance()
            cal.add(Calendar.DAY_OF_YEAR, -1)
            return dateFormat.format(cal.time)
        }

    val firstDayNextYearAndLastDayBeforeYear: Array<String?>
        get() {
            val datas = arrayOfNulls<String>(2)
            val dateFormat = SimpleDateFormat("yyyy-MM-dd", Locale.ENGLISH)
            var cal = Calendar.getInstance()
            cal.add(Calendar.YEAR, 1)
            cal.set(Calendar.DAY_OF_MONTH, 1)
            cal.set(Calendar.MONTH, 0)
            datas[0] = dateFormat.format(cal.time)
            cal = Calendar.getInstance()
            cal.add(Calendar.YEAR, -1)
            cal.set(Calendar.DAY_OF_MONTH, 31)
            cal.set(Calendar.MONTH, 11)
            datas[1] = dateFormat.format(cal.time)
            return datas
        }

    fun getCurrentDataWithAddSetMonthValue(first_data: String, add_month: Int): String {
        val dateFormat = SimpleDateFormat("yyyy-MM-dd", Locale.ENGLISH)
        val cal = Calendar.getInstance()
        if (first_data != "")
        //add first_data in calendar object to correct asign in second data
        {
            cal.set(Integer.parseInt(first_data.substring(0, 4)),
                    Integer.parseInt(first_data.substring(5, 7)) - 1,
                    Integer.parseInt(first_data.substring(8, 10)))
        }

        cal.add(Calendar.MONTH, add_month)  //Add two months to first data
        return dateFormat.format(cal.time)

    }

    fun getCurrentDataWithAddSetBeforeDays(first_data: String, addDays: Int): String {
        val cal = Calendar.getInstance()
        if (first_data != "")
        //add first_data in calendar object to correct asign in second data
        {

            cal.set(Integer.parseInt(first_data.substring(0, 4)),
                    Integer.parseInt(first_data.substring(5, 7)) - 1,
                    Integer.parseInt(first_data.substring(8, 10)))
        }

        cal.add(Calendar.DAY_OF_WEEK, -addDays)  //DownDays value to first data
        return SimpleDateFormat("yyyy-MM-dd", Locale.ENGLISH).format(cal.time)
    }

    fun getAllDataTimeCalendarObject(date: String, hour_string: String): Calendar {
        //Extract details data
        val year = Integer.parseInt(date.substring(0, 4))
        val month = Integer.parseInt(date.substring(5, 7)) - 1
        val day = Integer.parseInt(date.substring(8, 10))

        val hour_int = getHourMinutesSecondsInfo(hour_string)

        println(year.toString() + "/" + month + "/" + day + " " + hour_int[0] + ":" + hour_int[1] + ":" + hour_int[2])
        val calendar = Calendar.getInstance()
        calendar.set(year, month, day, hour_int[0], hour_int[1], hour_int[2])
        return calendar
    }

    fun getHourMinutesSecondsInfo(hour_string: String): IntArray {
        val hour = hour_string.substring(0, 2)
        val hourInt: Int
        if (hour[0] == '0') {
            hourInt = Integer.parseInt(hour.substring(1))
        } else {
            hourInt = Integer.parseInt(hour)
        }

        val minInt = Integer.parseInt(hour_string.substring(3, 5))
        return intArrayOf(hourInt, minInt, 0)
    }

    fun setFormatData(year: Int, monthOfYear: Int, dayOfMonth: Int): Array<String?> {
        //year/month/day
        val formatData = arrayOfNulls<String>(3)

        formatData[0] = year.toString()
        if (monthOfYear + 1 < 10) {
            formatData[1] = "0" + (monthOfYear + 1).toString()
        } else {
            formatData[1] = (monthOfYear + 1).toString()
        }

        if (dayOfMonth < 10) {
            formatData[2] = "0" + dayOfMonth.toString()
        } else {
            formatData[2] = dayOfMonth.toString()
        }
        return formatData
    }
}
